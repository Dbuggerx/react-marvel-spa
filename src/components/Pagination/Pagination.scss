@import '../scss/variables';
@import '../scss/placeholders';

.pagination {
  font-size: 0.9em;
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  width: 30%;

  @media #{map-get($breakpoints, medium)} {
    width: 70%;
  }

  @media #{map-get($breakpoints, small)} {
    width: 100%;
  }

  &__icon-defs {
    display: none;
  }

  &__prev-button {
    @extend %unstyled-button;
  }

  &__next-button {
    @extend %unstyled-button;
    transform: scale(-1, 1);
  }
}
