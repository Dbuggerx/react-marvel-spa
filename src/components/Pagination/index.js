// @flow

import React from 'react';
import type { StatelessFunctionalComponent } from 'react';
import './Pagination.scss';

export type Props = {
    currentPage: number,
    pageCount: number,
    goToPreviousPage: () => void,
    goToNextPage: () => void,
};

/**
 * Renders a pagination section
 */
const Pagination: StatelessFunctionalComponent<Props> = (props: Props) => (
    <React.Fragment>
        <svg width="0" height="0" viewBox="0 0 8 12">
            <defs>
                <path id="chevron-left" d="M 7.41,10.58 2.83,6 7.41,1.41 6,0 0,6 6,12 Z" />
            </defs>
        </svg>
        <nav className="pagination" aria-label="Pagination Navigation">
            <button
                className="pagination__prev-button"
                onClick={props.goToPreviousPage}
                aria-label="Go to previous page"
            >
                <svg viewBox="0 0 8 12">
                    <use xlinkHref="#chevron-left" />
                </svg>
            </button>
            Showing page {props.currentPage} of {props.pageCount}
            <button className="pagination__next-button" onClick={props.goToNextPage} aria-label="Go to next page">
                <svg viewBox="0 0 8 12">
                    <use xlinkHref="#chevron-left" />
                </svg>
            </button>
        </nav>
    </React.Fragment>
);

Pagination.displayName = 'Pagination';

export default Pagination;
