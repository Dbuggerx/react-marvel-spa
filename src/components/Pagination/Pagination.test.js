import React from 'react';
import { shallow } from 'enzyme';
import Pagination from './index';

describe('Pagination', () => {
    let wrapper;
    let goToPreviousPageMock;
    let goToNextPageMock;

    beforeEach(() => {
        goToPreviousPageMock = jest.fn();
        goToNextPageMock = jest.fn();

        wrapper = shallow(<Pagination
            currentPage={1}
            pageCount={10}
            goToPreviousPage={goToPreviousPageMock}
            goToNextPage={goToNextPageMock}
        />);
    });

    test('renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('back button calls "props.goToPreviousPage" upon click', () => {
        wrapper.find('[aria-label="Go to previous page"]').simulate('click');
        expect(goToPreviousPageMock).toHaveBeenCalledTimes(1);
    });

    test('forward button calls "props.goToNextPage" upon click', () => {
        wrapper.find('[aria-label="Go to next page"]').simulate('click');
        expect(goToNextPageMock).toHaveBeenCalledTimes(1);
    });
});
