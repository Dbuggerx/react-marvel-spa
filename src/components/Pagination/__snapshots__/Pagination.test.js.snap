// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders properly 1`] = `
<Fragment>
  <svg
    height="0"
    viewBox="0 0 8 12"
    width="0"
  >
    <defs>
      <path
        d="M 7.41,10.58 2.83,6 7.41,1.41 6,0 0,6 6,12 Z"
        id="chevron-left"
      />
    </defs>
  </svg>
  <nav
    aria-label="Pagination Navigation"
    className="pagination"
  >
    <button
      aria-label="Go to previous page"
      className="pagination__prev-button"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 8 12"
      >
        <use
          xlinkHref="#chevron-left"
        />
      </svg>
    </button>
    Showing page 
    1
     of 
    10
    <button
      aria-label="Go to next page"
      className="pagination__next-button"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 8 12"
      >
        <use
          xlinkHref="#chevron-left"
        />
      </svg>
    </button>
  </nav>
</Fragment>
`;
